### PR TITLE
wine: update to 5.22

### PR DESCRIPTION
**Pull request by eatradish**
Topic Description
-----------------

wine: update to 5.22

- Use SRCS to fetch wine-staging patches

Package(s) Affected
-------------------

wine 5.22

Security Update?
----------------

No

Architectural Progress
----------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

----

After the pull request is merged, all package(s) affected must be rebuilt against the `stable` Git tree and environment (only `stable` repository should be enabled in `sources.list`). This section marks the progress above.

Please, make sure the list of architectures below matches the ones above.

Post-Merge Architectural Progress
---------------------------------

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

### CODE REVIEW

**Review by MingcongBai — APPROVED** [review]

File: extra-emulation/wine/autobuild/prepare [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-git checkout v"${PKGVER}" || exit 1
-
 abinfo "Applying Wine Staging patches ..."
-cd "$SRCDIR"/wine-staging/patches
-ln -s ../../ wine
-./patchinstall.sh DESTDIR=../../ --all
-cd "$SRCDIR"/
+cd "$SRCDIR"/../wine-staging/patches
+ln -s "$SRCDIR" wine
+./patchinstall.sh DESTDIR="$SRCDIR" --all
```
Comment (MingcongBai) [inline extra-emulation/wine/autobuild/prepare]: Please use absolute path (`"$SRCDIR"/patchinstall.sh`) here.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: extra-emulation/wine/autobuild/prepare (hunk 1 above)
Comment (eatradish) [reply extra-emulation/wine/autobuild/prepare]: Done

File: extra-emulation/wine/autobuild/prepare [hunk 2]
```diff
@@ -1,13 +1,8 @@
...
-cd "$SRCDIR"/wine-staging
-git checkout v"${PKGVER}" || exit 1
-
 abinfo "Applying Wine Staging patches ..."
-cd "$SRCDIR"/wine-staging/patches
-ln -s ../../ wine
-./patchinstall.sh DESTDIR=../../ --all
-cd "$SRCDIR"/
+cd "$SRCDIR"/../wine-staging/patches
+ln -s "$SRCDIR" wine
```
Comment (MingcongBai) [inline extra-emulation/wine/autobuild/prepare]: Verbose and non-absolute path detected...

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Comment by eatradish** [comment]
image

LGTM

**Pull request merged**